include/boost_plugin_loader/utils.h

```diff
@@ -36,11 +38,13 @@ class PluginLoaderException : public std::runtime_error
 
 /**
  * @brief Attempt to load library give library name and directory
+ * @param ec The error code to determine if the library was successfully loaded
  * @param library_name The library name to load which does not include the prefix 'lib' or suffix '.so'
  * @param library_directory The library directory, if empty it will enable search system directories
  * @return A shared library
  */
-boost::dll::shared_library loadLibrary(const std::string& library_name, const std::string& library_directory = "");
+boost::dll::shared_library loadLibrary(boost::dll::fs::error_code& ec, const std::string& library_name,
+                                       const std::string& library_directory = "");
```

Comment:
```suggestion
boost::dll::shared_library loadLibrary(const std::string& library_name,
                                       const std::string& library_directory = "", boost::dll::fs::error_code* ec = nullptr);
```

I would think about a way of changing this function to maintain backwards compatibility. It seems like we could get the same updated behavior with the above proposed change and still maintain backwards compatibility

Reply:
This started to look ugly so I did some cleanup to make it backwards compatible and marked a few function deprecated.